src/twisted/web/test/test_wsgi.py

```diff
@@ -1941,6 +1943,91 @@ def cbRendered(ignored):
         return d
 
 
+    @inlineCallbacks
+    def test_pendingRequestOperationsRespectConnectionLoss(self):
+        """
+        If there are pending transport writes after
+        C{Request.notifyFinish} is called, the transport should
+        not be written to. The test issues a request to a wsgi
+        app, which schedules the transport to disconnect. All writes
+        to the transport are blocked until after the disconnect
+        finishes, forcing the write-after-disconnect behaviour.
+        """
+
+        self.enableThreads()
+        trackedRequest = [None]
+
+        def applicationFactory():
+            """
+            A wsgi application which schedules a request
+            disconnect, and returns a single response chunk.
+            """
+            def application(environ, startResponse):
+                startResponse('200 OK', [])
+
+                self.reactor.callFromThread(
+                    trackedRequest[0].connectionLost,
+                    Failure(ConnectionLost("No more connection")))
+
+                yield b'single chunk response'
+
+            return application
+
+        def requestFactory(*args, **kwargs):
+            # A factory which returns a twisted.web.server.Request
+            # and stores a reference to it in trackedRequest[0].
+
+            # Self.lowLevelRender() is call only once, and it
+            # is expected to call requestFactory() once.
+            self.assertIsNone(trackedRequest[0])
+            trackedRequest[0] = Request(*args, **kwargs)
+            return trackedRequest[0]
+
+        _originalWSGIResponseWrite = _WSGIResponse.write
+
+        # Event signalled after trackedRequest[0] is
+        # disconnected.
+        connectionLostSyncPoint = threading.Event()
+        wsgiResponseWriteResult = Deferred()
+
+        def _WSGIResponse_write(self, data):
+            """
+            A wrapper for C{_WSGIResponse.write} which first
+            blocks until C{connectionLostSyncPoint} is
+            signalled, and puts the result/failure
+            in C{writeResult}.
+            @return: None
+            """
+            connectionLostSyncPoint.wait()
+
+            try:
+                r = _originalWSGIResponseWrite(self, data)
+                self.reactor.callFromThread(
+                    wsgiResponseWriteResult.callback, None)
```

Comment:
minor comment. maybe instead of "None" we should pass "data" to the wsgiResponseWriteResult and then we can check what we got here.
I expect that is little place for failure, but if its not too hard I prefer to assert as much as possible when dealing with patched/rigged objects.

Reply:
Replaced `None` with `data` and checked for equality at the end of the test.